arwen/host/execution.go

```diff
@@ -191,9 +191,11 @@ func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmO
 
 	// Perform a value transfer to the called SC. If the execution fails, this
 	// transfer will not persist.
-	err = output.TransferValueOnly(input.RecipientAddr, input.CallerAddr, input.CallValue)
-	if err != nil {
-		return
+	if input.CallType != vmcommon.AsynchronousCallBack || input.CallValue.Cmp(arwen.Zero) == 0 {
```

Comment:
This condition prevents in-shard callbacks from receiving any eGLD?

Reply:
in case of callBacks the transfer is done beforehand. as the contract has to call transferValue explicitily before.